File: play-2.4.x/K-Modules/kmodules/K-Base/app/models/K/EbeanConfig/NewTaskPersistController.java
package models.K.EbeanConfig;


import com.avaje.ebean.event.BeanPersistController;
import com.avaje.ebean.event.BeanPersistRequest;
import models.K.BgTask.PlanTask;
import play.Logger;

import java.util.Set;

/**
 * Created by kk on 15/1/20.
 */
public class NewTaskPersistController implements BeanPersistController {

    public NewTaskPersistController() {
        Logger.debug("Ebean server load NewTaskPersistController");
    }

    @Override
    public int getExecutionOrder() {
        return 11;
    }

    @Override
    public boolean isRegisterFor(Class<?> cls) {
        String bean_cls_name = cls.getName();
        if (bean_cls_name.equals(PlanTask.class.getName())) {
            return true;
        }
        return false;
    }

    @Override
    public boolean preInsert(BeanPersistRequest<?> request) {
        return true;
    }

    @Override
    public boolean preUpdate(BeanPersistRequest<?> request) {
        return true;
    }

    @Override
    public boolean preDelete(BeanPersistRequest<?> request) {
        return true;
    }

    @Override
    public void postInsert(BeanPersistRequest<?> request) {
        String bean_cls_name = request.getBean().getClass().getName();
//        Logger.debug("New Task: {}", Helper.ToJsonStringPretty(request.getBean()));
        if (bean_cls_name.equals(PlanTask.class.getName())) {
            request.getTransaction().putUserObject(PlanTask.class.getName(), "");
        }
    }

    @Override
    public void postUpdate(BeanPersistRequest<?> request) {
        // do nothing
    }

    @Override
    public void postDelete(BeanPersistRequest<?> request) {
        // do nothing
    }

    @Override
    public void postLoad(Object bean, Set<String> includedProperties) {
        // do nothing
    }
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/models/K/EbeanConfig/NewTaskPersistController.kt
package models.K.EbeanConfig


import K.Common.Helper
import com.avaje.ebean.event.BeanPersistController
import com.avaje.ebean.event.BeanPersistRequest
import models.K.BgTask.PlanTask

/**
 * Created by kk on 15/1/20.
 */
class NewTaskPersistController : BeanPersistController {

    init {
        Helper.DLog("Ebean server load NewTaskPersistController")
    }

    override fun getExecutionOrder(): Int {
        return 11
    }

    override fun isRegisterFor(cls: Class<*>): Boolean {
        val bean_cls_name = cls.name
        if (bean_cls_name == PlanTask::class.java.name) {
            return true
        }
        return false
    }

    override fun preInsert(request: BeanPersistRequest<*>): Boolean {
        return true
    }

    override fun preUpdate(request: BeanPersistRequest<*>): Boolean {
        return true
    }

    override fun preDelete(request: BeanPersistRequest<*>): Boolean {
        return true
    }

    override fun postInsert(request: BeanPersistRequest<*>) {
        val bean_cls_name = request.bean.javaClass.name
        //        Logger.debug("New Task: {}", Helper.ToJsonStringPretty(request.getBean()));
        if (bean_cls_name == PlanTask::class.java.name) {
            request.transaction.putUserObject(PlanTask::class.java.name, "")
        }
    }

    override fun postUpdate(request: BeanPersistRequest<*>) {
        // do nothing
    }

    override fun postDelete(request: BeanPersistRequest<*>) {
        // do nothing
    }

}
